### ISSUE TITLE
Required Dimensionality for `pos_weight` conflicts with Documentation on Multi-Dim Labels

### ISSUE THREAD

**Issue by elliesch** [pytorch/pytorch#82496]
### 🐛 Describe the bug

The documentation for the required shape of `pos_weight` in [`BCEwithLogitsLoss`](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html?highlight=bce#torch.nn.BCEWithLogitsLoss) suggests a 1D vector is required:

> pos_weight ([Tensor](https://pytorch.org/docs/stable/tensors.html#torch.Tensor), optional) – a weight of positive examples. Must be a vector with length equal to the number of classes.

In the case where class labels are themselves >1D, however, `pos_weight` requires a tensor with as many dimensions as the labels. I.e. if there are 3 classes of labels that have 2 dimensions each, the shape of `pos_weight` needs to be `([3, None, None])` and a tensor with shape `([3]),` which is the length of the number of classes, will throw an error. (See [this stack overflow question](https://stackoverflow.com/questions/73159568/multi-channel-2d-mask-weights-using-bcewithlogitsloss-in-pytorch/73163289#73163289) for more discussion on this.)

### Versions

PyTorch version: 1.12.0+cu102
Is debug build: False
CUDA used to build PyTorch: 10.2
ROCM used to build PyTorch: N/A

OS: Ubuntu 22.04 LTS (x86_64)
GCC version: (conda-forge gcc 9.4.0-16) 9.4.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.35

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 16:56:21)  [GCC 10.3.0] (64-bit runtime)
Python platform: Linux-5.4.188-104.359.amzn2.x86_64-x86_64-with-glibc2.35
Is CUDA available: True
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Tesla T4
Nvidia driver version: 470.57.02
cuDNN version: Could not collect
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] torch==1.12.0
[pip3] torch-lr-finder==0.2.1
[pip3] torchsummary==1.5.1
[pip3] torchvision==0.13.0
[conda] cudatoolkit               11.7.0              hd8887f6_10    conda-forge
[conda] efficientnet-pytorch      0.6.3              pyh9f0ad1d_0    conda-forge
[conda] libblas                   3.9.0            15_linux64_mkl    conda-forge
[conda] libcblas                  3.9.0            15_linux64_mkl    conda-forge
[conda] liblapack                 3.9.0            15_linux64_mkl    conda-forge
[conda] magma                     2.5.4                h6103c52_2    conda-forge
[conda] mkl                       2022.1.0           h84fe81f_915    conda-forge
[conda] mkl-service               2.4.0            py39hb699420_0    conda-forge
[conda] numpy                     1.22.1           py39h91f2184_0    conda-forge
[conda] numpydoc                  1.4.0              pyhd8ed1ab_1    conda-forge
[conda] pytorch                   1.10.2          cuda112py39h4de5995_1    conda-forge
[conda] pytorch-gpu               1.10.2          cuda112py39h0bbbad9_1    conda-forge
[conda] pytorch-lightning         1.6.5              pyhd8ed1ab_0    conda-forge
[conda] segmentation-models-pytorch 0.2.1              pyhd8ed1ab_0    conda-forge
[conda] torchgeo                  0.3.0              pyhd8ed1ab_0    conda-forge
[conda] torchmetrics              0.9.3              pyhd8ed1ab_0    conda-forge
[conda] torchvision               0.11.3          cuda112py39ha475c55_2    conda-forge

cc @svekars @holly1238 @albanD @mruberry @jbschlosser @walterddr @kshitij12345 @saketh-are

**Comment by jbschlosser** [comment]
Hey @elliesch, thanks for the report! We'd accept a PR correcting the expected shape(s) for `pos_weight` in the docs.

Related: #65702

If you'd like to contribute, it should be pretty easy to knock both of these out at once :)
